### PR TITLE
chore(batch): Housekeeping for recent changes

### PR DESCRIPTION
**Pull request by michaelbrewer**
**Issue #, if available:**

## Description of changes:

Changes:
* base.py - `records` can't just be iterable as we reference it by `__getitem__`
* base.py - logger.debugger is expecting a `%s`
* docs - Typos in the docs
* formatter.py - `_build_root_keys` can be a static method
* typing - add some missing unit tests

**Checklist**

 

* [X] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [X] Update tests
* [X] Update docs
* [X] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws_lambda_powertools/utilities/batch/base.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def failure_handler(self, record: Any, exception: Exception):
             "fail", exceptions args, original record
         """
         entry = ("fail", exception.args, record)
-        logger.debug("Record processing exception: ", exception)
+        logger.debug("Record processing exception: %s", exception)
```
Comment (heitorlessa) [inline aws_lambda_powertools/utilities/batch/base.py:98]: Could you change to f-string for consistency, please?

File: aws_lambda_powertools/utilities/typing/lambda_cognito_identity.py [hunk 2]
```diff
@@ -2,9 +2,7 @@
 
 
 class LambdaCognitoIdentity(object):
-    """
-    Information about the Amazon Cognito identity that authorized the request.
-    """
+    """Information about the Amazon Cognito identity that authorized the request."""
```
Comment (heitorlessa) [inline aws_lambda_powertools/utilities/typing/lambda_cognito_identity.py:5]: This reminds me we need to prioritise a minter for NumPy docstrings

**Review by heitorlessa — CHANGES_REQUESTED** [review]
Thanks for the housekeeping help before 1.5.0 - One minor on string interpolation for consistency.

File: aws_lambda_powertools/utilities/batch/base.py (hunk 1 above)
Comment (michaelbrewer) [reply aws_lambda_powertools/utilities/batch/base.py:98]: @heitorlessa  sure.

File: aws_lambda_powertools/utilities/typing/lambda_cognito_identity.py (hunk 2 above)
Comment (michaelbrewer) [reply aws_lambda_powertools/utilities/typing/lambda_cognito_identity.py:5]: @heitorlessa - sure

File: aws_lambda_powertools/utilities/batch/base.py (hunk 1 above)
Comment (michaelbrewer) [reply aws_lambda_powertools/utilities/batch/base.py]: @heitorlessa  done.

**Review by heitorlessa — APPROVED** [review]

**Pull request merged**